tsl/src/continuous_aggs/invalidation.c

```diff
@@ -313,27 +313,43 @@ invalidation_hyper_log_add_entry(int32 hyper_id, int64 start, int64 end)
  * hypertable.
  */
 void
-invalidation_add_entry(const Hypertable *ht, ContinuousAggHypertableStatus caggstatus,
-					   int32 entry_id, int64 start, int64 end)
+continuous_agg_invalidate(const Hypertable *raw_ht, int64 start, int64 end)
 {
-	Assert(ht != NULL);
+	continuous_agg_invalidate_with_materialization(raw_ht, NULL, start, end);
+}
+
+void
+continuous_agg_invalidate_with_materialization(const Hypertable *raw_ht, const Hypertable *mat_ht,
+											   int64 start, int64 end)
+{
+	ContinuousAggHypertableStatus caggstatus = ts_continuous_agg_hypertable_status(raw_ht->fd.id);
```

Comment:
This only gets the status of `raw_ht`, when it should get the status of `mat_ht` for materialized hypertables.

Reply:
The old code always get the status of this first parameter that is always the `RawHypertable`. If you look below I force the status to by `HypertableIsMaterialization` when we pass the `mat_ht` parameter. The old code path does exactly it but by passing down the status and entry_id (can be the id of `mat_ht` or the `raw_ht`).

And if this new code path is really broken then our regression tests is also broken, no?

Reply:
That is correct, the regression tests are also broken.
You only check `(mat_ht)` for the distributed case, in the normal case you never go to `case HypertableIsMaterialization:`.

If you look at line `src/process_utility.c:1035`:
```
						ts_cm_functions
							->continuous_agg_invalidate_with_materialization(raw_ht,
																			 mat_ht,
																			 TS_TIME_NOBEGIN,
																			 TS_TIME_NOEND);
```
This will never work for normal hypertables.